components/tag/stories/tag.stories.js

```diff
@@ -130,26 +129,233 @@ export default {
 			type: "migrated",
 		},
 	},
+	decorators: [
+		(Story, context) => html`
+			<style>
+				.spectrum-Detail { display: inline-block; }
+				.spectrum-Typography > div {
+					border: 1px solid var(--spectrum-gray-200);
+					border-radius: 4px;
+					padding: 0 1em 1em;
+					/* Why seafoam? Because it separates it from the component styles. */
+					--mod-detail-font-color: var(--spectrum-seafoam-900);
+				}
+			</style>
+			<div
+				style=${styleMap({
+					display: "flex",
+					flexDirection: "column",
+					alignItems: "flex-start",
+					gap: "1rem",
+					"--mod-detail-margin-end": ".3rem",
+				})}
+			>
+				${Story(context)}
+			</div>
+		`,
+	],
 };
 
-export const Default = Template.bind({});
-Default.args = {};
+const Tags = (args) => html` <div
+	style=${styleMap({
+		display: "flex",
+		gap: "1rem",
+	})}
+>
+	${Template({...args})}
+	${when(window.isChromatic(), () =>
+		Template({
+			...args,
+			label: "Tag with longer content that truncates",
+		})
+	)}
+	${when(window.isChromatic(), () =>
+		Template({
+			...args,
+			visual: "Icon",
+		})
+	)}
+	${when(window.isChromatic(), () =>
+		Template({
+			...args,
+			visual: "Avatar",
+		})
+	)}
+	${when(window.isChromatic(), () =>
+		Template({
+			...args,
+			visual: "Thumbnail",
+		})
+	)}
+</div>`;
 
-export const Icon = Template.bind({});
-Icon.args = {
-	hasIcon: true,
-	iconName: "Info",
-	label: "Tag label that truncates when it gets too long",
-	customStyles: {"max-inline-size": "200px"}
-};
+const States = (args) =>
+	html` <div>
+			${Typography({
+				semantics: "detail",
+				size: "m",
+				content: ["Default"],
+			})}
+			${Tags(args)}
+		</div>
+		<div>
+			${Typography({
+				semantics: "detail",
+				size: "m",
+				content: ["Selected"],
+			})}
+			${Tags({
+				...args,
+				isSelected: true,
+			})}
+		</div>
+		<div>
+			${Typography({
+				semantics: "detail",
+				size: "m",
+				content: ["Disabled"],
+			})}
+			${Tags({
+				...args,
+				isDisabled: true,
+			})}
+		</div>
+		<div>
+			${Typography({
+				semantics: "detail",
+				size: "m",
+				content: ["Disabled & Selected"],
+			})}
+			${Tags({
+				...args,
+				isDisabled: true,
+				isSelected: true,
+			})}
+		</div>`;
 
-export const Avatar = Template.bind({});
-Avatar.args = {
-	hasAvatar: true,
-	avatarUrl: "example-ava.png"
-};
+const Sizes = (args) =>
+	html` ${["s", "m", "l"].map((size) => {
+		return html` <div>
+			${Typography({
+				semantics: "detail",
+				size: "s",
+				content: [
+					{
+						xxs: "Extra-extra-small",
+						xs: "Extra-small",
+						s: "Small",
+						m: "Medium",
+						l: "Large",
+						xl: "Extra-large",
+						xxl: "Extra-extra-large",
+					}[size],
```

Comment:
Do we need all these sizes, or just small, medium, and large? 🤔

Reply:
I copied this from button/action button, which also don't have as wide a spread of sizes as there are options listed here, and wondered the same thing. I tend to agree and would err on the side of removing the extra sizes unless there's a compelling reason to keep them--I'll plan to take them out.